tests/net/buf_simple/src/main.c

```diff
@@ -0,0 +1,406 @@
+
+	zassert_mem_equal(be24, net_buf_simple_pull_mem(&buf, sizeof(be24)),
+			  sizeof(be24), "Invalid 24 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_pull_le32)
+{
+	net_buf_simple_add_mem(&buf, &le32, sizeof(le32));
+
+	zassert_equal(u32, net_buf_simple_pull_le32(&buf),
+		      "Invalid 32 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_pull_be32)
+{
+	net_buf_simple_add_mem(&buf, &be32, sizeof(be32));
+
+	zassert_equal(u32, net_buf_simple_pull_be32(&buf),
+		      "Invalid 32 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_add_le32)
+{
+	net_buf_simple_add_le32(&buf, u32);
+
+	zassert_mem_equal(le32, net_buf_simple_pull_mem(&buf, sizeof(le32)),
+			  sizeof(le32), "Invalid 32 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_add_be32)
+{
+	net_buf_simple_add_be32(&buf, u32);
+
+	zassert_mem_equal(be32, net_buf_simple_pull_mem(&buf, sizeof(be32)),
+			  sizeof(be32), "Invalid 32 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_pull_le48)
+{
+	net_buf_simple_add_mem(&buf, &le48, sizeof(le48));
+
+	zassert_equal(u48, net_buf_simple_pull_le48(&buf),
+		      "Invalid 48 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_pull_be48)
+{
+	net_buf_simple_add_mem(&buf, &be48, sizeof(be48));
+
+	zassert_equal(u48, net_buf_simple_pull_be48(&buf),
+		      "Invalid 48 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_add_le48)
+{
+	net_buf_simple_add_le48(&buf, u48);
+
+	zassert_mem_equal(le48, net_buf_simple_pull_mem(&buf, sizeof(le48)),
+			  sizeof(le48), "Invalid 48 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_add_be48)
+{
+	net_buf_simple_add_be48(&buf, u48);
+
+	zassert_mem_equal(be48, net_buf_simple_pull_mem(&buf, sizeof(be48)),
+			  sizeof(be48), "Invalid 48 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_pull_le64)
+{
+	net_buf_simple_add_mem(&buf, &le64, sizeof(le64));
+
+	zassert_equal(u64, net_buf_simple_pull_le64(&buf),
+		      "Invalid 64 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_pull_be64)
+{
+	net_buf_simple_add_mem(&buf, &be64, sizeof(be64));
+
+	zassert_equal(u64, net_buf_simple_pull_be64(&buf),
+		      "Invalid 64 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_add_le64)
+{
+	net_buf_simple_add_le64(&buf, u64);
+
+	zassert_mem_equal(le64, net_buf_simple_pull_mem(&buf, sizeof(le64)),
+			  sizeof(le64), "Invalid 64 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_add_be64)
+{
+	net_buf_simple_add_be64(&buf, u64);
+
+	zassert_mem_equal(be64, net_buf_simple_pull_mem(&buf, sizeof(be64)),
+			  sizeof(be64), "Invalid 64 bits byte order");
+}
+
+ZTEST(net_buf_simple_test_suite, test_net_buf_simple_remove_le16)
+{
+	net_buf_simple_reserve(&buf, 16);
```

Comment:
Why are we reserving 16 octets? 

IMO we should either reserve the number of octets we want to push/remove, or we should use the buffer's size to reserve all (which is what the code is doing now)

Reply:
Yeah, I agree, but it does not make much difference for the test. It was just copied from `tests/net/buf`